src/test/java/net/minecraftforge/debug/StructureGenTest.java

```diff
@@ -0,0 +1,205 @@
+    }
+
+    @SubscribeEvent
+    public void onCollectStructuresOverworld(InitStructureGensEvent<ChunkGeneratorOverworld> event)
+    {
+        logger.info("Collecting structures for Overworld");
+        event.addStructure(new TestStructure("overworld"));
+    }
+
+    @SubscribeEvent
+    public void onCollectStructuresEnd(InitStructureGensEvent<ChunkGeneratorEnd> event)
+    {
+        logger.info("Collecting structures for End");
+        event.addStructure(new TestStructure("end"));
+    }
+
+    @SubscribeEvent
+    public void onCollectStructuresHell(InitStructureGensEvent<ChunkGeneratorHell> event)
+    {
+        logger.info("Collecting structures for Hell");
+        event.addStructure(new TestStructure("hell"));
+    }
+
+    @SubscribeEvent
+    public void onCollectStructures(InitStructureGensEvent event)
+    {
+        logger.info("Collecting structures for {}", event.getGenericType());
+    }
+
+    /**
+     * Has to be public due to the MapGenStructureIO system
+     */
+    public static class Start extends StructureStart
+    {
+        Start(int x, int z)
+        {
+            super(x, z);
+            this.updateBoundingBox();
+        }
+
+        /**
+         * Required. Don't use. Used internally
+         */
+        public Start()
+        {
+            super();
+        }
+    }
+
+    private class TestStructure extends MapGenStructure implements MapGenStructureManager.ISpawningStructure
+    {
+
+        private final String generator;
+        private boolean spawn = true;
+        private boolean isAt = true;
+        private boolean generate = true;
+        private boolean generateStructure = true;
+
+        private TestStructure(String generator)
+        {
+            this.generator = generator;
+        }
+
+        @Override
+        public String getStructureName()
+        {
+            return "forge-test-structure";
+        }
+
+        @Override
+        public void generate(World worldIn, int x, int z, ChunkPrimer primer)
+        {
+            if (generate)
+            {
+                logger.info("Generating {}", generator);
+                generate = false;
+            }
+            super.generate(worldIn, x, z, primer);
+        }
+
+        @Override
+        public synchronized boolean generateStructure(World worldIn, Random randomIn, ChunkPos chunkCoord)
+        {
+            if (generateStructure)
+            {
+                logger.info("Generating structure {}", generator);
+                generateStructure = false;
+            }
+            return super.generateStructure(worldIn, randomIn, chunkCoord);
+        }
+
+        @Nullable
+        @Override
+        public BlockPos getNearestStructurePos(World worldIn, BlockPos pos, boolean findUnexplored)
+        {
+            logger.info("Getting nearest structure {}", generator);
+            if (isComplete())
+            {
+                FMLLog.bigWarning("Structure Gen Test in {} successful", generator);
```

Comment:
use info if it's not an error

Reply:
but it is inside a (disabled by default) testmod and usually only triggered by a command, so I think that isn't that big a deal and it is much easier to spot during testing

Reply:
It's just bad practice, people use the test mods as coding examples sometimes.
I'll definitely see an info log if I'm looking for it, no problem.

Reply:
Agreed, changed